ADA/databricks_workflows.qmd

```diff
@@ -0,0 +1,104 @@
+    \
+    Any libraries or packages that the workflow depends on can be specified which will ensure that these are installed on the cluster when it begins (not applicable to SQL warehouses).\
+    \
+    You are also able to set task level parameters here which mean that you are able to reuse the same notebook for different tasks by setting different inputs. This allows you to create notebooks that serve as 'functions' from R or Python, or 'stored procedures' from SQL.\
+    \
+    Here you can also setup notifications for the specific task. In addition you can specify the number of times a task should be retried if it fails, and a duration threshold for the task before forcing it to fail.
+
+![](/images/ada-workflow-task.png){fig-align="center"}
+
+::: callout-note
+## Source
+
+The source by default is set to your Workspace, but it is recommended that you use a version controlled Git repository instead. This prevents you changing the code of a notebook during a workflow run as the tasks are sourced from a specific repository version, branch, commit or tag rather than a workbook you may be working on.
+:::
+
+::: callout-note
+## Compute
+
+During active development it is probably easiest to use your personal compute or a SQL warehouse you have access to.
+
+Once development is complete and the workflow becomes business as usual it may be worth requesting a specific job cluster to be created by the [ADA team](adaptteam@education.gov.uk) to be solely responsible for running the workflow.
+:::
+
+5.  Click the 'Create task' button to save the details. After this you will be able to add additional tasks.
+6.  Once you begin creating a second task all tasks will now have 2 additional configurations; 'depends on', and 'run if dependencies'.\
+    \
+    **Depends on** - a list of tasks that must be run before the start of this task\
+    **Run if dependencies** - Instructions for the conditions to run the task based on the dependencies set above. The default option is 'All succeeded' but there are also the following options:\
+    \
+    -   At least one succeeded\
+    -   None failed\
+    -   All done\
+    -   At least one failed\
+    -   All failed\
+7.  After setting up a flow of tasks you will be presented with a graphical presentation of the workflow as seen below:
+
+![](/images/ada-workflow-task-chart.png)
+
+------------------------------------------------------------------------
+
+### Auditing
+
+------------------------------------------------------------------------
+
+Each time a workflow is ran DataBricks audits any input parameters, all outputs, the success and failure of each task, along with when it was run and who by.
+
+This makes them a very powerful debugging tool as you can refer back to results from previous runs.
```

Comment:
Have expanded on this sentence a bit in the latest commit, included a bit about repairing failed workflows as well because I forgot about that feature when writing this.